megatron/optimizer/clip_grads.py

```diff
@@ -110,8 +113,8 @@ def clip_grad_norm_fp32(parameters, max_norm, norm_type=2):
 
     # Scale.
     clip_coeff = max_norm / (total_norm + 1.0e-6)
-    if clip_coeff < 1.0:
-        dummy_overflow_buf = torch.cuda.IntTensor([0])
+    if get_accelerator().device_name() == 'cuda' and clip_coeff < 1.0:
```

Comment:
This is a problem as it makes gradient clipping available only for cuda. I think we need these two solutions: 
1. Python based gradient clipping (slower).
2. Accelerator specific gradient clipping. 

This PR should include (1) at the minimum so that (2) can be added as needed for different accelerators.

Reply:
added python based implementation.